### DISCUSSION TITLE
Physics-based animation

### DISCUSSION THREAD

**Discussion by sammonius** [godotengine/godot-proposals#6819] [General Discussions]
I think it would be nice to have a node similar to the AnimationPlayer, but instead of rotating and positioning obects, it adds forces to RigidBodies in order to play the animation. This would be useful in physics-based games, like for example a boxing game where you want players to be able to bounce around and fall.

Each object (bone) could have a fallback mechanism for if outside forces interfere with the animation:

1. Hault the rest of the animation until the interfered object catches up
2. Apply a stronger force to have it catch up without haulting the animation (how much inertia does the interfering force have?)
3. Kick the object out of the animation, like a branch falling off of a tree
4. Ignore, and continue the animation out of sync

It would be especially nice to use with PhysicalBone objects to create moving ragdolls. It's fun to imagine what could be made with something like this.

**Comment by AThousandShips** [comment]
You can already do this partially with method tracks but having the ability to do it more dynamically is interesting

**Reply by AThousandShips** [reply]
Do however keep in mind that physics based animation won't be deterministic due to the non-deterministic nature of the physics engine, this still affects animation on physics bodies but important to keep in mind

**Comment by InTheProcess** [comment]
I wish this was in Godot!  This feature would be amazing!

**Reply by Calinou** [reply]
Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.
